compiler/src/iree/compiler/Utils/TracingUtils.cpp

```diff
@@ -35,3 +35,63 @@ void PassTracing::runAfterPassFailed(Pass *pass, Operation *op) {
 
 }  // namespace iree_compiler
 }  // namespace mlir
+
+#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_ALLOCATION_TRACKING
+
+// Replace most forms of the allocation and deallocation functions.
+
+void *operator new(std::size_t count) {
+  auto ptr = malloc(count);
+  IREE_TRACE_ALLOC(ptr, count);
+  return ptr;
+}
+void *operator new[](std::size_t count) {
+  auto ptr = malloc(count);
+  IREE_TRACE_ALLOC(ptr, count);
+  return ptr;
+}
+void *operator new(std::size_t count, std::align_val_t al) {
+  auto ptr = malloc(count);
+  IREE_TRACE_ALLOC(ptr, count);
+  return ptr;
+}
+void *operator new[](std::size_t count, std::align_val_t al) {
+  auto ptr = malloc(count);
+  IREE_TRACE_ALLOC(ptr, count);
+  return ptr;
+}
```

Comment:
I think we need to honor the alignment here and below as otherwise we may get crashes

https://www.cppstories.com/2019/08/newnew-align/#custom-overloads has some examples we can follow (including msvc/win handling!)

Reply:
I tried that code and didn't see any crashes... but when I try turning ASan on, I see this error:
```
==2872958==ERROR: AddressSanitizer: invalid alignment requested in aligned_alloc: 8, alignment must be a power of two and the requested size 0x15 must be a multiple of alignment (thread T0)
    #0 0x2d0132 in aligned_alloc (/usr/local/google/home/scotttodd/code/iree-build/tools/iree-compile+0x2d0132)
    #1 0x7f79f4bf319c in operator new(unsigned long, std::align_val_t) /usr/local/google/home/scotttodd/code/iree/compiler/src/iree/compiler/Utils/TracingUtils.cpp:80:14
    #2 0x7f79f88b49ac in llvm::allocate_buffer(unsigned long, unsigned long) /usr/local/google/home/scotttodd/code/iree/third_party/llvm-project/llvm/lib/Support/MemAlloc.cpp:16:10
```

(I'm a bit surprised that ASan works at all - I'm tempted to just `#error` if ASan/MSan/TSan are enabled)